sklearn/cluster/_hdbscan/tests/test_hdbscan.py

```diff
@@ -0,0 +1,311 @@
+"""
+Tests for HDBSCAN clustering algorithm
+Based on the DBSCAN test code
+"""
+import numpy as np
+import pytest
+from scipy import sparse, stats
+from scipy.spatial import distance
+
+from sklearn.cluster import HDBSCAN
+from sklearn.datasets import make_blobs
+from sklearn.metrics import fowlkes_mallows_score
+from sklearn.metrics.pairwise import _VALID_METRICS
+from sklearn.neighbors import BallTree, KDTree
+from sklearn.preprocessing import StandardScaler
+from sklearn.utils import shuffle
+from sklearn.utils._testing import assert_array_almost_equal
+
+n_clusters_true = 3
+X, y = make_blobs(n_samples=200, random_state=10)
+X, y = shuffle(X, y, random_state=7)
+X = StandardScaler().fit_transform(X)
+
+X_missing_data = X.copy()
+X_missing_data[0] = [np.nan, 1]
+X_missing_data[5] = [np.nan, np.nan]
+
+
+def test_missing_data():
+    """
+    Tests if nan data are treated as infinite distance from all other points
+    and assigned to -1 cluster.
```

Comment:
Is `np.inf` also treated as infinite distances? If so can we parameterize and test that?